NorthstarDLL/audio.cpp

```diff
@@ -15,6 +15,15 @@ extern "C"
 {
 	// should be called only in LoadSampleMetadata_Hook
 	extern void* __fastcall Audio_GetParentEvent();
+
+#ifdef __MINGW32__
+	void* __attribute__((noinline)) Audio_GetParentEvent()
+	{
+		void* r12;
+		asm volatile("mov {%%r12, %[reg]|%[reg], r12}" : [reg] "=r"(r12));
+		return r12;
+	}
+#endif
```

Comment:
Why is this inlined here?

Reply:
The existing assembly can't compile under gcc and since gcc supports inline assembly, it was simpler to write the replacement that way.